packages/dara-core/tests/python/test_websocket.py

```diff
@@ -188,3 +194,229 @@ def handler(loop, context):
+                    assert second_ws_message.get('message').get('variable').get('__typename') == 'Variable'
+                    assert second_ws_message.get('message').get('variable').get('default') == 'current'
+
+                    # send a response back
+                    await second_websocket.send_json({
+                        'type': 'message',
+                        'channel': second_ws_message.get('message').get('__rchan'),
+                        'message': NOT_REGISTERED
+                    })
+
+                async with anyio.create_task_group() as tg_second:
+                    tg_second.start_soon(client)
+                    tg_second.start_soon(second_client)
+                    tg_second.start_soon(second_server)
+                
+                # Assert that the second variable is correctly set based on the original value as the second value is not registered
+                assert second_var_value == expected_return_value
+
+            # Assert that the first value remains unaffected by the second value
+            assert var_value == expected_return_value
+
+async def test_two_websockets_only_one_with_value_return_exact():
+    app, token = setup_two_websocket_tests()
+
+    async with AsyncTestClient(app) as testclient:
+        async with _async_ws_connect(testclient, token) as websocket:
+            init = await websocket.receive_json()
+            variable = Variable(default='current')
+            expected_return_value = 'return_value'
+            var_value = None
+
+            async def server():
+                nonlocal var_value
+                var_value = await variable.get_current_value()
+
+            async def client():
+                # get msg from server
+                websocket_msg = await websocket.receive_json()
+
+                # Assert that the request is correct
+                assert websocket_msg.get('message').get('variable').get('__typename') == 'Variable'
+                assert websocket_msg.get('message').get('variable').get('default') == 'current'
+
+                # send a response back
+                await websocket.send_json({
+                    'type': 'message',
+                    'channel': websocket_msg.get('message').get('__rchan'),
+                    'message': expected_return_value
+                })
+
+            async with anyio.create_task_group() as tg:
+                tg.start_soon(client)
+                tg.start_soon(server)
+
+            async with _async_ws_connect(testclient, token) as second_websocket:
+                second_init = await second_websocket.receive_json()
+                second_var_value = None
+                async def second_server():
+                    nonlocal second_var_value
+                    second_var_value = await variable.get_current_value()
+
+                ## Make sure the websocket channel is set correctly
+                WS_CHANNEL.set(second_init.get('message').get('channel'))
```

Comment:
Can we have another test checking that if we don't set this variable, a dict is returned with both values?